static/app/components/dropdownMenu.tsx

```diff
@@ -105,43 +105,51 @@ type Props = DefaultProps & {
   children: (renderProps: RenderProps) => React.ReactNode;
 };
 
-type State = {
-  isOpen: boolean;
-};
-
-class DropdownMenu extends React.Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    keepMenuOpen: false,
-    closeOnEscape: true,
-  };
-
-  state: State = {
-    isOpen: false,
-  };
-
-  componentWillUnmount() {
-    document.removeEventListener('click', this.checkClickOutside, true);
-  }
-
-  dropdownMenu: Element | null = null;
-  dropdownActor: Element | null = null;
-
-  mouseLeaveId: number | null = null;
-  mouseEnterId: number | null = null;
+function DropdownMenu({
+  keepMenuOpen = false,
+  closeOnEscape = true,
+  onOpen,
+  onClose,
+  onClickOutside,
+  shouldIgnoreClickOutside,
+  isOpen: isOpenProp,
+  alwaysRenderMenu,
+  isNestedDropdown,
+  children,
+}: Props) {
+  const dropdownMenu = useRef<Element | null>(null);
+  const dropdownActor = useRef<Element | null>(null);
+
+  let mouseLeaveId: number | null = null;
+  let mouseEnterId: number | null = null;
```

Comment:
You probably want both of these to be refs as well I think.

Reply:
Otherwise every render they will be re-assigned.

Reply:
👍🏼 for ref or state, else these will both be lost on rerender